File: jpa-save/src/main/java/io/github/hyeonic/jpasave/member/Member.java
package io.github.hyeonic.jpasave.member;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Member {

    @Id
    @GeneratedValue
    private Long id;

    private String name;

    protected Member() {
    }

    public Member(final Long id, final String name) {
        this.id = id;
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public void setName(final String name) {
        this.name = name;
    }
}

File: spring-data-envers/src/main/kotlin/hyeoni/c/springdataenvers/domain/member/Member.kt
package hyeoni.c.springdataenvers.domain.member

import org.hibernate.envers.Audited
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Audited
@Entity
class Member(
    name: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    @Column
    var name: String = name
        protected set

    override fun toString(): String {
        return "Member(id=$id, name='$name')"
    }
}
